docs/project/moderators.md

```diff
@@ -0,0 +1,101 @@
+# Moderators
+
+<!--
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+-->
+
+## Overview
+
+Moderation is primarily focused on Carbon's Discord and GitHub, but generally
+applies to the [collaboration systems](/CONTRIBUTING.md#collaboration-systems).
+Moderators are empowered to improve community discussion and enforce the
+[Code of Conduct](/CODE_OF_CONDUCT.md). Conduct decisions are ultimately the
+domain of the [Conduct team](/CODE_OF_CONDUCT.md#conduct-team), although
+moderators will often act in order to protect the community.
+
+All moderators will have the "moderator" role on Discord. Some moderators are
+_senior moderators_, and will have extra privileges to help address harmful
+conduct. Senior moderators will have the "senior-moderator" role on Discord.
+
+## Becoming a moderator
+
+TODO: Need to figure out right process for asking. (that is, how?)
+
+The Conduct team reviews new moderators. Community members will be considered
+mainly based on their history on the Carbon Language Discord, although
+applicants may also note history on other servers.
+
+Prior to being granted moderation privileges, new moderators will be join a
+workshop in order to help orient them with the Carbon Language community's
+values and moderator expectations.
+
+## Promotion to senior moderator
+
+The Conduct team reviews promotions to senior moderator. Moderators will be
+considered for promotion to senior moderators when:
+
+-   They have demonstrated that they will be active moderators.
+-   They are making decisions consistent with the
+    [Code of Conduct](/CODE_OF_CONDUCT.md).
+-   Senior moderators are taking actions which require senior moderator
+    privileges at the moderator's request.
+
+## Moderation powers
+
+There is a moderation playbook for moderators with instructions on how to use
+these powers.
+
+-   **Discord**
+
+    -   **Delete message** (senior): Delete a message owned by another user.
+
```

Comment:
Just me, but I'd generally expect all of the "delete message" and "edit message" capabilities to be roughly the same tier. Below, they are "all", which I think make sense.

The "senior" capabilities would then be specifically ones with a larger scope of impact -- entire conversation, channel, or persistent user effect.

Reply:
You're not specific so I assume you're not familiar with GitHub's ACL controls. Unlike Discord, GitHub is more granular with permissions. As far as I know, the permissions like "Lock conversation" or "Hide message" cannot be granted separately from "Delete conversation" or "Delete message".

So here are what I see as the options:

1. Only allow _senior_ moderators permissions on GitHub. Regular moderators will not have any special permissions.
2. Grant all moderators permission to delete messages on Discord.
3. Leave this as is.

I think what you're suggesting is closest to (1), requiring regular moderators to ask for a senior moderator to help with issues on GitHub. Is that your intent?

Reply:
No, the granularity wasn't at all clear to me and that led to my suggestion. If there is no ability to have different controls between locking and hiding (non-destructive) and editing and deleting (destructive), not sure I have strong opinions about which option we use. I'll leave it to you and other moderators.


Regardless, maybe change the structure of the GitHub section to make it clear that there's really just a single control here that covers all of these operations? I had assumed the bullet points here roughly reflected the granularity that these roles can be set up.

Reply:
> locking and hiding (non-destructive) and editing and deleting (destructive),

On some further thought, hiding looks separate, so I've reduced to that at least. I don't think locking is an option.

> Regardless, maybe change the structure of the GitHub section to make it clear that there's really just a single control here that covers all of these operations?

Added a note beneath the permissions; I hadn't intended this page to be documenting the implementation, mainly the consequences. i.e., more a reference for moderators than for admin.